pytensor/tensor/math.py

```diff
@@ -2841,6 +2841,138 @@ def matmul(x1: "ArrayLike", x2: "ArrayLike", dtype: Optional["DTypeLike"] = None
     return out
 
 
+def vecdot(
+    x1: "ArrayLike",
+    x2: "ArrayLike",
+    axis: int = -1,
```

Comment:
I was surprised by this, didn't know vecdot had axis and in the numpy docstrings it's buried so it may be just something cropping up from the ufunc. Not sure we want to add it, would need to check numpy manually.

Also the original tests (didn't check yet in this PR) weren't comparing with vecdot which was suspicious ?